lib/route.js

```diff
@@ -1,6 +1,7 @@
 'use strict'
 
 const FindMyWay = require('find-my-way')
+const equal = require('fast-deep-equal')
```

Comment:
Not sure if it supports the custom constraint strategies:

- https://github.com/delvedor/find-my-way/#custom-constraint-strategies
- this package simplify its usage: https://github.com/Eomm/header-constraint-strategy/

we should add a test for this use case and - if it is not supported by `equal` I think we need a simplified comparison function that check the `name` property when a custom strategy is provided